test/lib/auth.js

```diff
@@ -64,7 +64,156 @@ describe('Auth lib', function () {
+	    appId: config.oauth2.clientID,
+	    user: {
+		id: 'rick',
+		nickName: 'theMagician',
+		accessToken: 'accessToken',
+		refreshToken: 'refreshToken',
+		organizations: [{
+		    id: '123456789',
+		    name: 'patty',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '987654321',
+		    name: 'MntyPythn',
+		    roles: [{
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '111555999',
+		    name: 'AmaneceQueNoEsPoco',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '1',
+			name: 'purchuaser'
+		    }]
+		}]
+	    }
+        };
+
+	var header =  {
+	    'x-organization': '123456789'
+	};
+
+	it ('should continue with middleware processing if no user object has been provided', function (done) {
+	    var req = {
+		id: 'test_user',
+		appId: config.oauth2.clientID,
+            };
+	    auth.checkOrganizations(req, {}, done);
+	});
+	    
+	it('Function should return unauthorized if header has been set but that org isnt on the organization list', function(done) {
+
+	    var resExp = {};
+	    var msgExp = 'Not allowed to access the provided organization';
+	    
+	    var unauthorized = getUnauthorizedMock(resExp, msgExp, done);
+	    
+	    var auth = getAuthLib(strategy, {}, unauthorized, {});
+	    
+	    var req = {
+		id: 'test_user',
+		appId: config.oauth2.clientID,
+		user: {
+		    id: 'rick',
+		    nickName: 'theMagician',
+		    accessToken: 'accessToken',
+		    refreshToken: 'refreshToken',
+		    organizations: []
+		},
+		headers: {
+		    'x-organization': '111555999'
+		}
+	    };
+	    
+	    auth.setPartyObj(req, resExp, done);
+	});
+
+	it('Individual object should be selected if x-organization header not present', function(done) {
+	    var expectedUser = {
+		id: 'rick',
+		nickName: 'theMagician',
+		accessToken: 'accessToken',
+		refreshToken: 'refreshToken',
+		organizations: [{
+		    id: '123456789',
+		    name: 'patty',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '987654321',
+		    name: 'MntyPythn',
+		    roles: [{
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '111555999',
+		    name: 'AmaneceQueNoEsPoco',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '1',
+			name: 'purchuaser'
+		    }]
+		}]
+	    };
+	    auth.setPartyObj(reqInd, {}, function() {
+		expect(reqInd.user).toEqual(expectedUser);
+		done();
+	    });
+	    
+	});
+
+	it('Organization object should be selected if x-organization header is present', function(done) {
+	    var expectedUser = {
+		userNickname: 'rick',
+		id: '123456789',
+		roles: [{
+		    id: '3',
+		    name: 'Seller'
+		}, {
+		    id: '6',
+		    name: 'provider'
+		}],
+		displayName: 'patty',
+		accessToken: 'accessToken',
+		refreshToken: 'refreshToken',
+		email: '123456789' + '@emailnotusable.com'
+	    };
+
+	    reqInd.headers = header;
```

Comment:
doing this you are overriding the global reqInd. If for any reason the previous test is executed after this one, it is going to fail